src/tools/tracejob.c

```diff
@@ -403,15 +405,21 @@ parse_log(FILE *fp, char *job, int ind)
 	strcpy(job_buf, job);
 
 	while (fgets(buf, buf_size, fp) != NULL) {
-		while(buf_size == strlen(buf) + 1){
-			buf = (char*)realloc(buf, buf_size * sizeof(char*));
-			tbuf = (char*)realloc(tbuf, buf_size * sizeof(char*));
-			if(buf == NULL || tbuf == NULL || fgets(tbuf, buf_size+1, fp) == NULL){
+		while(buf_size == strlen(buf) + incr){
+			buf_size *= 2;
+			buf = (char*)realloc(buf, buf_size + 1 * sizeof(char));
```

Comment:
You need a set of ()'s around buf_size * 1.  Right now you're reallocing to buf_size + (1*sizeof(char)).  It just so happens that since the size of a char = 1, that it all works out.